### PR TITLE
Add more system tracing to SDK startup

### PR DESCRIPTION
**Pull request by bidetofevil**
## Goal

Add more detail system tracing info to the SDK startup process. Pulled the module names out so they are defined in a single place, per review comment.

### CODE REVIEW

**Comment by bidetofevil** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/427?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #425** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/425?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #426** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/426?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #427** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/427?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
        * **PR #428** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/428?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/427?utm_source=stack-comment).

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/injection/ModuleInitBootstrapper.kt [hunk 1]
```diff
@@ -430,4 +461,32 @@ internal class ModuleInitBootstrapper(
...
     private fun isInitialized(): Boolean = asyncInitTask.get() != null
+
+    private fun <T> init(module: KClass<*>, provider: Provider<T>): T =
+        Systrace.traceSynchronous("${moduleNames[module] ?: "module"}-init") { provider() }
+
+    private fun <T> postInit(module: KClass<*>, code: () -> T): T =
+        Systrace.traceSynchronous("${moduleNames[module] ?: "module"}-post-init") { code() }
+
+    companion object {
+        private val moduleNames: Map<KClass<*>, String> = mapOf(
```
Comment (fractalwrench) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/injection/ModuleInitBootstrapper.kt:472]: Could `Class#simpleName` be an alternative to this? That way we don't have to maintain a fairly brittle collection of names in production code

**Review by fractalwrench — APPROVED** [review]
LGTM

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/injection/ModuleInitBootstrapper.kt (hunk 1 above)
Comment (bidetofevil) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/injection/ModuleInitBootstrapper.kt:472]: Hmm. yeah, I guess obfuscation doesn't matter eh? Let me change that

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/injection/ModuleInitBootstrapper.kt [hunk 2]
```diff
@@ -430,4 +464,14 @@ internal class ModuleInitBootstrapper(
...
+
+    private fun <T> init(module: KClass<*>, provider: Provider<T>): T =
+        Systrace.traceSynchronous("${toSectionName(module)}-init") { provider() }
+
+    private fun <T> postInit(module: KClass<*>, code: () -> T): T =
+        Systrace.traceSynchronous("${toSectionName(module)}-post-init") { code() }
+
+    // This is called twice for each input - memoizing/caching is not worth the hassle
+    private fun toSectionName(klass: KClass<*>): String =
+        klass.simpleName?.removeSuffix("Module")?.toLowerCase(Locale.ENGLISH) ?: "module"
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/injection/ModuleInitBootstrapper.kt:476]: Note: we already have it in our proguard files to not obfuscate our classes, so I don't even have to worry about that!

**Comment by bidetofevil** [comment]
### Merge activity

* **Feb 16, 5:11 PM EST**: @bidetofevil started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/427).

**Pull request merged**